### PR TITLE
Add validation for username, firstname, lastname, and password fields

### LINKED ISSUE #64: Add Validation for Username, Firstname, Lastname, and Password Fields
# Issue Report

## Title

Add Validation for Username, Firstname, Lastname, and Password Fields

## Description

Currently, when creating an account, the system allows registration with just one character in the username, firstname, and lastname fields. Additionally, there is no password validation to ensure a strong password. I propose adding validation for these fields:

- Username, Firstname, and Lastname Validation: These fields should have a minimum of 2 characters, and should not allow digits in the firstname or lastname.
- Password Validation: The password should meet the following criteria for strength:
- At least 8 characters
- Contains at least one digit
- Contains at least one special character

## Steps to Reproduce the Issue

1. Attempt to register an account with only one character in the username, firstname, or lastname fields. It allows the registration.
2. The system also allows digits in the firstname and lastname.
3. No password validation is in place for ensuring strong passwords.

## Expected Behavior

The system should enforce a minimum of 2 characters in the username, firstname, and lastname, disallowing digits in the firstname and lastname. For the password, it should enforce a minimum of 8 characters, including at least one digit and one special character.

## Actual Behavior

Currently, the form allows registration with just one character in the username, firstname, and lastname, and it also permits digits in the firstname and lastname fields. Additionally, weak passwords are allowed without any validation for length, digits, or special characters.

## Screenshots (if applicable)
Screenshot 2024-10-03 115253

## Additional Context
I propose implementing these validations to improve the security and data integrity of the registration process.

@saurabhbakolia Please assign this task to me.

### PR DESCRIPTION
**Pull request by prakrutipathak**
# Pull Request Template

**Title:** Add validation for username, firstname, lastname, and password fields

## Description
This PR adds validation to the registration form to ensure proper input for the username, firstname, lastname, and password fields. Previously, users could register with just one character in these fields, and there was no password strength validation.

## Changes Made

- Added validation to ensure the username, firstname, and lastname fields have a minimum of 2 characters and do not allow digits.
- Implemented password validation requiring a minimum of 8 characters, with at least one digit and one special character.

##Related Issue
Fixes #64 

## Checklist

- [x] I have tested my code and it works as expected.
- [ ] I have added/updated documentation where necessary.
- [ ] I have included tests for new features.
- [x] I have followed the project's coding standards.

## Screenshots (if applicable)

https://github.com/user-attachments/assets/7662df49-5b2c-4678-a3e7-79d05be22e75

### CODE REVIEW

File: src/pages/AllProducts.jsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import React from 'react'
+import Navbar from '../Components/Navbar'
```
Comment (saurabhbakolia) [inline src/pages/AllProducts.jsx:2]: Take pull from `develop` branch and use lowercase `components`

File: src/pages/Register.jsx [hunk 2]
```diff
@@ -103,129 +103,119 @@ const PasswordBox = styled.div`
...
-								Already have an account? <Link href="/login">LOGIN</Link>
-							</Text>
-						</Box>
-						<Button type="submit">CREATE</Button>
-					</Form>
-				</Wrapper>
-			</Container>
-		</>
-	);
+    const [isLoading, setIsLoading] = useState(false);
```
Comment (saurabhbakolia) [inline src/pages/Register.jsx:106]: Use Biome to format

**Review by saurabhbakolia — CHANGES_REQUESTED** [review]

File: src/pages/AllProducts.jsx (hunk 1 above)
Comment (prakrutipathak) [reply src/pages/AllProducts.jsx]: @saurabhbakolia I have done changes can you please check it out

**Comment by saurabhbakolia** [comment]
@prakrutipathak resolve conflicts.

**Comment by prakrutipathak** [comment]
@saurabhbakolia Done can you check it?

**Review by saurabhbakolia — APPROVED** [review]

**Review by saurabhbakolia — APPROVED** [review]

**Review by saurabhbakolia — APPROVED** [review]

**Pull request merged**